astropy/units/utils.py

```diff
@@ -247,12 +247,12 @@ def validate_power(p):
         try:
             p = float(p)
         except Exception:
-            if not np.isscalar(p):
+            if not np.size(p) == 1:
```

Comment:
I think in princple the intent here was to test that `p` was a true scalar, so in that sense the code was fine... But, really, we could be more general and check whether the array is actually filled with the same values (the problem is with having *different* powers, not really with the shape), so how about
```
try:
    return float(p)
except Exception:
    try:
        p = np.asanyarray(p)
        if ((first := p.flat[0]) == p).all():
            return float(first)
    except Exception:
        pass

raise ValueError("Quantities and Units may only be raised to a scalar power")
```

Reply:
Oh, I like that much better, good idea!